### PR TITLE
Bump Go to 1.21.1

### PR DESCRIPTION
**Pull request by C0rWin**

### CODE REVIEW

**Comment by denyeart** [comment]
Thanks @C0rWin , it looks like a unit test needs to be updated to match updated error messages in Go 1.21.

The Broken Link Checker failure can be ignored (it only gets run when you update the broken-link-checker.yml file).

**Comment by C0rWin** [comment]
> Thanks @C0rWin , it looks like a unit test needs to be updated to match updated error messages in Go 1.21.
> 
> The Broken Link Checker failure can be ignored (it only gets run when you update the broken-link-checker.yml file).

yep, @denyeart will update unit-test

File: .github/workflows/broken-link-checker.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ on:
       - .github/workflows/broken-link-checker.yml
 
 env:
-  GO_VER: 1.20.7
+  GO_VER: 1.21.0
```
Comment (yeasy) [inline .github/workflows/broken-link-checker.yml]: We can update to 1.21.1 directly now.

**Comment by C0rWin** [comment]
There are a number of CFT-Deliver tests that are failing in CI, while I cannot spot similar failures locally, e.g. cannot reproduce them. 🤔

**Comment by denyeart** [comment]
I've re-run the failed unit test check, and it keeps failing in the same place:

```
Summarizing 2 Failures:
  [FAIL] CFT-Deliverer the consecutive errors are unbounded and the peer is not a static leader [It] hits the maximum sleep time value in an exponential fashion and retries until exceeding the max retry duration
  /home/runner/work/fabric/fabric/internal/pkg/peer/blocksprovider/deliverer_test.go:278
  [FAIL] CFT-Deliverer the consecutive errors are unbounded and the peer is static leader [It] hits the maximum sleep time value in an exponential fashion and retries indefinitely
  /home/runner/work/fabric/fabric/internal/pkg/peer/blocksprovider/deliverer_test.go:294

Ran 28 of 28 Specs in 2.390 seconds
FAIL! -- 26 Passed | 2 Failed | 0 Pending | 0 Skipped
--- FAIL: TestBlocksProvider (2.39s)
FAIL
coverage: 77.4% of statements
FAIL	github.com/hyperledger/fabric/internal/pkg/peer/blocksprovider	2.775s
```

Like @C0rWin I can't reproduce when testing locally with Go 1.21.1, the tests pass on my mac.

I don't understand the failure in CI. @tock-ibm can you take a look and provide your thoughts? I think you are familiar with these tests...

**Comment by C0rWin** [comment]
> I've re-run the failed unit test check, and it keeps failing in the same place:
> 
> ```
> Summarizing 2 Failures:
>   [FAIL] CFT-Deliverer the consecutive errors are unbounded and the peer is not a static leader [It] hits the maximum sleep time value in an exponential fashion and retries until exceeding the max retry duration
>   /home/runner/work/fabric/fabric/internal/pkg/peer/blocksprovider/deliverer_test.go:278
>   [FAIL] CFT-Deliverer the consecutive errors are unbounded and the peer is static leader [It] hits the maximum sleep time value in an exponential fashion and retries indefinitely
>   /home/runner/work/fabric/fabric/internal/pkg/peer/blocksprovider/deliverer_test.go:294
> 
> Ran 28 of 28 Specs in 2.390 seconds
> FAIL! -- 26 Passed | 2 Failed | 0 Pending | 0 Skipped
> --- FAIL: TestBlocksProvider (2.39s)
> FAIL
> coverage: 77.4% of statements
> FAIL	github.com/hyperledger/fabric/internal/pkg/peer/blocksprovider	2.775s
> ```
> 
> Like @C0rWin I can't reproduce when testing locally with Go 1.21.1, the tests pass on my mac.
> 
> I don't understand the failure in CI. @tock-ibm can you take a look and provide your thoughts? I think you are familiar with these tests...

Both tests are fairly simple they just make sure that backoff behaves as expected and assert later the amount of sleep calls to see whenever backoff stops if the condition met or keeps trying.

**Comment by C0rWin** [comment]
Hey, @denyeart, I managed to make it pass. The tests were too slow and lacked time to complete, hence the error.

**Review by denyeart — APPROVED** [review]

**Pull request merged**